src/inmanta/ast/entity.py

```diff
@@ -381,7 +381,16 @@ def add_to_index(self, instance: Instance) -> None:
         Update indexes based on the instance and the attribute that has
         been set
         """
-        attributes = {k: repr(v.get_value()) for (k, v) in instance.slots.items() if v.is_ready()}
+        attributes = {}
+        for k, v in instance.slots.items():
+            if v.is_ready():
+                value = v.get_value()
+                # Check if the value is an integer,
+                # and convert to float if necessary
+                if isinstance(value, int):
+                    value = float(value)
```

Comment:
We should not convert all ints to floats, only if the attribute type is float.

Reply:
And use `Float.cast` to handle `Unknown`

Reply:
Actually, we don't need the index construction patch, do we? I don't really understand why you changed the index approach. The goal was to "keep `float` as in the PR".

Reply:
@sanderr I'm not sure I understand this comment. Do you mean that noting needs to be changed about the indexes? I thought this was the original goal?

Reply:
because If I can't change the number type and I don't change the way indexes are added, I don't see how I can make this test pass:
```

def test_int_as_index_for_number(snippetcompiler):
    snippetcompiler.setup_for_snippet(
        f"""
entity A:
    number x
end
implement A using std::none
index A(x)
test = (A(x=0) == A(x=0.0))
test = true
        """,
    )
    compiler.do_compile()

```
This will always add 2 different indexes

Reply:
I removed this test. It passes with floats but not with the number type

Reply:
> It passes with floats but not with the number type

Perfect. We wanted to "keep float as in the PR" and "keep number as in the current stable but deprecate it. Do not apply any fixes for it, like the index matching."